### ISSUE TITLE
Doc issue and some code

### ISSUE THREAD

**Issue by deg** [jebberjeb/fsmviz#3]
Thank you for this library, and your excellent blog posts on using FSMs in re-frame.

I started playing with this and hit a minor doc issue:  In your readme, you discuss using this in Clojure/ClojureScript and in JavaScript. The wording, as I first read it, implied that both the map and tuple form were ok in ClojureScript, and only direct usage from JavaScript was limited to the tuple form.

Of course, as I realized a minute later, this was not true. ClojureScript also needs the tuple form.
A few lines of clarification in the readme would be great!

I also wrote a few lines to handle ClojureScript. I was hoping to offer a PR once I had it cleaned up, but looks like I'm going to be heading down a side path for a while, because my app needs something closer to simple state charts.  So, figured I'd toss the code to you here.  Feel free to include, discard, or wait for me to eventually maybe offer some cleanups.

```
(defn- flatten-graph
  "Convert state graph to tuple form"
  [graph]
  (reduce-kv
   (fn [coll source edge-map]
     (into coll
           (reduce-kv
            (fn [sub-coll edge sink]
              (conj sub-coll [(name source) (name edge) (name sink)]))
            []
            edge-map)))
   []
   graph))

;; [TODO] Move this into a re-frame sub
(defn render-graph
  "generate-image returns the SVG as an XML string with an embedded explicit width
  and height. Here we use an ugly hack to strip this info, and then embed it into
  hiccup for reagent."
  [graph]
  [:div {:dangerouslySetInnerHTML
         {:__html (-> graph
                      flatten-graph
                      (fsmviz.core/generate-image "fsm")
                      (clojure.string/replace-first #"width=\"\d*pt\""  "")
                      (clojure.string/replace-first #"height=\"\d*pt\"" ""))}}])

```

Side note: the limitation I hit is that I need multiple logged-in states, but wanted an easy and DRY way to test for being logged in. I'm still playing with this, but the basic idea is to keep a stack of states, and support `:shift`, `:push`, and `:pop` state transitions.  I'm only a few minutes into this idea, but it feels like it will play out cleanly.

**Comment by jebberjeb** [comment]
Thanks for reporting this, taking a look today. The map form should definitely work for Clojurescript. I'm wondering if I broke that when adding the JS API during my most recent changes. I'll definitely take a closer look today.

**Comment by deg** [comment]
Ok, thanks. My apologies that that I didn't save the failure I saw. I don't recall exactly what it was, but think it was one of the usual generic Clojure "you gave me something in the wrong structure" barfs. This, of course, led me to assume that I needed to flatten the graph.

Meanwhile, you may also be interested in how I've extended the FSM to handle modal states.  See https://github.com/deg/trilystro/blob/master/src/cljs/trilystro/fsm.cljs and its image, attached here.
![trilystro-fsm](https://user-images.githubusercontent.com/90383/31585210-dde6db30-b1c6-11e7-98d6-1bfadd063754.png)

**Comment by deg** [comment]
FWIW, the problem is between `transform-js-data` which returns a vector, and the `::fsm-map` which expects a map in this case.

I'm not sure enough of your use cases to offer a PR; sorry. For my case, `transform-js-data` is not needed.

**Comment by jebberjeb** [comment]
Thanks @deg. I just recreated it w/ in cljs w/ a simple project and am working on a fix.

**Comment by jebberjeb** [comment]
Fixed by Release 0.1.3. The README has been updated w/ docs on how to run an Clojurescript example which contains both the map and tuple syntax.
